pkg/api/apiserver.go

```diff
@@ -0,0 +1,62 @@
+package api
+
+import (
+	"crypto/sha256"
+	"crypto/subtle"
+	"fmt"
+	"net/http"
+
+	"github.com/vshn/vshn-sli-reporting/pkg/api/downtime"
+	"github.com/vshn/vshn-sli-reporting/pkg/store"
+)
+
+type ApiServerConfig struct {
+	AuthUser string
+	AuthPass string
+	Port     int
+	Host     string
+}
+
+type ApiServer struct {
+	config ApiServerConfig
+	mux    *http.ServeMux
+	store  store.DowntimeStore
+}
+
+func NewApiServer(config ApiServerConfig, store store.DowntimeStore) ApiServer {
+	var mux = http.NewServeMux()
+	downtime.Setup(mux, store)
+	return ApiServer{
+		config: config,
+		mux:    mux,
+		store:  store,
+	}
+
+}
+
+func (s *ApiServer) Start() error {
```

Comment:
We migth want to implement a graceful shutdown by either passing a context oder adding a `.Stop()` or `.Close()` method.

Reply:
There we go, that should do the trick